src/lib/y2partitioner/widgets/delete_disk_partition_button.rb

```diff
@@ -48,7 +52,42 @@ def handle
 
     private
 
+      def confirm(device)
+        names = children_names(device)
+
+        if names.empty?
+          # TODO: check for children that will die and if there are, use confirm_recursive_delete
+          Yast::Popup.YesNo(
+            # TRANSLATORS %s is device to be deleted
+            format(_("Really delete %s?"), device.name)
+          )
+        else
+          confirm_recursive_delete(
+            names,
+            _("Confirm Deleting of All Partitions"),
+            format(_("The %{type} \"%{name}\" contains at least one another device.\n" \
+              "If you proceed, the following devices will be deleted:"),
+              name: device.name,
+              type: device.is?(:disk) ? _("disk") : _("partition")),
```

Comment:
Once again, remember this doesn't include DASD. Maybe you want to consider to add a `:disk_device` (or similar) handler for `Disk#is?` and `Dasd#is?` so you can ask `device.is?(:disk_device)`. Kind of symmetric to [`Devicegraph#disk_devices](https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/devicegraph.rb#L155)